apptail/storage/gob_handler.go

```diff
@@ -0,0 +1,79 @@
+package storage
+
+import (
+	"bytes"
+	"encoding/gob"
+	"github.com/ActiveState/log"
+	"io/ioutil"
+	"os"
+)
+
+// exposing these for testing
+type Storage interface {
+	Encode(data interface{}) ([]byte, error)
+	Load(data interface{})
+	Write(buf []byte) error
+}
+
+type FileStorage struct {
+	file_path string
+}
+
+const FILE_MODE = 0666
+
+func NewFileStorage(path string) Storage {
+	return &FileStorage{
+		file_path: path,
+	}
+
+}
+
+func (s *FileStorage) Encode(data interface{}) ([]byte, error) {
+	m := new(bytes.Buffer)
+	enc := gob.NewEncoder(m)
+	err := enc.Encode(data)
+	if err != nil {
+		return nil, err
+
+	}
+	return m.Bytes(), nil
+}
+
+func (s *FileStorage) Write(buf []byte) error {
+
+	if err := ioutil.WriteFile(s.file_path, buf, FILE_MODE); err != nil {
+
+		return err
+
+	}
+	// this extra step to make the file accessible by stackato user
+	if err := os.Chmod(s.file_path, FILE_MODE); err != nil {
+		return err
+
+	}
+	return nil
+}
+
+func (s *FileStorage) Load(e interface{}) {
```

Comment:
I feel like we should also return an `error` here (and have early returns when read / decode fails), and the caller can explicitly ignore it instead.  That way it's clearer that it was intentional.